qp_klp/klp.py

```diff
@@ -136,391 +50,43 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
-                    sample_name_diff = sheet_samples - qsamples
-
-            if sample_name_diff:
-                # before we report as an error, check tube_id
-                error_tube_id = 'No tube_id column in Qiita.'
-                if tube_id_present:
-                    tids = qclient.get(f'{qurl}/categories=tube_id')['samples']
-                    # generate a map of sample_names to tube_ids for
-                    # GenPrepFileJob.
-                    sn_tid_map_by_project[project_name] = {
-                        y[0]: x.replace(f'{qiita_id}.', '') for x, y in
-                        tids.items()}
-                    tids = set(sn_tid_map_by_project[project_name].keys())
-                    tube_id_diff = sheet_samples - tids
-                    if not tube_id_diff:
-                        continue
-                    len_tube_id_overlap = len(tube_id_diff)
-                    tids_example = ', '.join(choices(list(tids), k=5))
-                    error_tube_id = (
-                        f'tube_id in Qiita but {len_tube_id_overlap} missing '
-                        f'samples. Some samples from tube_id: {tids_example}.')
-
-                len_overlap = len(sample_name_diff)
-                # selecting at random k=5 samples to minimize space in display
-                samples_example = ', '.join(choices(list(qsamples), k=5))
-                # selecting the up to 4 first samples to minimize space in
-                # display
-                missing = ', '.join(sorted(sample_name_diff)[:4])
-                errors.append(
-                    f'{project} has {len_overlap} missing samples (i.e. '
-                    f'{missing}). Some samples from Qiita: {samples_example}. '
-                    f'{error_tube_id}')
-
-        if errors:
-            return False, None, '\n'.join(errors)
-
-        with open(sample_sheet_path, 'w') as f:
-            sheet.write(f)
-
-        # Create a Pipeline object
-        try:
-            pipeline = Pipeline(CONFIG_FP, run_identifier, sample_sheet_path,
-                                out_dir, job_id)
-        except PipelineError as e:
-            # Pipeline is the object that finds the input fp, based on
-            # a search directory set in configuration.json and a run_id.
-            if str(e).endswith("could not be found"):
-                msg = f"A path for {run_identifier} could not be found."
-                return False, None, msg
-            elif str(e).startswith("Sample-sheet has the following errors:"):
-                msg = str(e)
-                _update_current_message(msg)
-                raise ValueError(msg)
+    # StatusUpdate uses qclient to easily update the status-line and maintain
+    # a certain amount of state needed to provide the updates users are
+    # accustomed to. E.g. "Step 1 of 6: Setting up pipeline"
+    status_line = StatusUpdate(qclient, job_id)
+
+    try:
```

Comment:
This is an extremely long `try`, is it necessary? If yes, can you split in smaller blocks of try/except with specific exceptions in those smaller blocks? Using this kind of long try blocks hides what's actually raising the exceptions.

Reply:
Will do!